docs: move production "Backups" and "Upgrading" sections

drrosa opened the pull request:
Moves the production "Backups" and "Upgrading" sections from `maintain-secure-upgrade.md` to `export-and-import.md` and `modifying-zulip.md`.

Fixes #13184.

**Testing Plan:**
`./tools/test-documentation --skip-external-links`
`./tools/lint --groups=backend --skip=gitlint,mypy`

**GIFs or Screenshots:**
![image](https://user-images.githubusercontent.com/2343554/65559235-b44c6900-deee-11e9-9c27-d7b26b08c323.png)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zulip/zulip/13222)
<!-- Reviewable:end -->

drrosa commented:
@timabbott I think it's ready :)

drrosa commented:
I think the only thing missing now is renaming filenames and main headings to reflect changes.
Name suggestions:
`maintain-secure-upgrade.md` -> ` maintain-and-manage.md`
`modifying-zulip.md` -> `modify-and-upgrade.md`

We should probably leave file renames for last to avoid potential rebase issues. Let me know if everything else looks good though.

timabbott commented:
I think renaming `maintain-secure-upgrade` we may not want to do because there's a lot of existing links to subsection of that page (which are hard to redirect) from places like our release notes blog posts; I think we might want to actually create some sections that just have redirect links on that page, and eventually just have it be a tiny page with such links.  

`upgrade-or-modify` I think is right for that bundle, which I'm fine with making a new name since `modifying-zulip` hasn't had subsections linked to broadly as it's new.  "Upgrade" feels like it should come first since that's probably 50x more common.

drrosa commented:
Okay Done. There's an error I haven't seen before though:

`./tools/test-help-documentation --skip-external-links`
```bash
[help_documentation_crawler] ERROR: Please check link: <404 https://zulip.readthedocs.io/en/latest/production/upgrade-or-modify.html>
```

Looks like it is checking for the corresponding external link which doesn't exist yet.

timabbott commented:
Thanks for working on this @drrosa!  I did another round of simplifying the headings discussion of backups and merged that commit as dc5dbcbee382acc85d6a5051dd31c48eec9143da.  I have one follow-up request for that, which is that we should make a small section at the end of `maintain-secure-upgrade` that just has a link to the new place in the docs where Backups live, so any direct links to that part of the docs aren't useless.

For the upgrade/modify commit, the changes were not what I had in mind.  I wanted to keep the "Upgrading" section more or less verbatim as-is, and just add them on top of the "Modifying" section, in recognition of that being 100x more common and important; whereas your changes sorta reworked things to make the "modifying" stuff primary.  I've attempted to undo that (and pushed back to this PR with my current progress on doing headings etc.), with the article back to its original filename just to make the diffs easier to read.  I don't think I've succeeded fully in getting the order right, but have to go AFK right now.  I also got a bit carried away in making the first "Upgrade" docs section nicer, so be sure to base new work off this PR.

Additionally, the wrong set of sections moved; "Zulip and the API" should have stayed behind, and the "Upgrading the base OS" stuff should have moved.  

I think we also need links a high-level ToS links section similar to what was in "maintain, secure, upgrade" for the new page.

Can you do the next pass on this?

drrosa commented:
Okay, no problem. I'll do the next pass

drrosa commented:
- [x] Added ToS links for the upgrading/modifying sections
- [x] Added the "Upgrading the operating system" section that should've been moved
- [x] Removed "API and your Zulip URL" and "Memory leak mitigation" that should've stayed behind
- [x] Added "Backups, export and import" to the related content links section in maintain-secure-upgrade, below the ToS.

timabbott commented:
Great, thanks @drrosa!  I did another fairly large round of changes to this and merged it as af4d7b4b52c5af46bc9cb9f61d88c0d9ecf2fc4f.

I think the new page is pretty reasonable, though it could probably use a read.  We still have a lot of work to do on the now quite messy "maintain, secure, upgrade" page to get to a releaseable checkpoint.  I'll start a thread on what to do there in #documentation.
